### PR TITLE
[cxx-interop] Mark un-specialized class templates as unavailable in Swift.

### PR DESCRIPTION
**Pull request by zoecarver**
They don't really work, so let's not "support" them yet.

### CODE REVIEW

**Comment by zoecarver** [comment]
Draft because I haven't fixed the tests yet.

**Comment by zoecarver** [comment]
@swift-ci please test

**Comment by zoecarver** [comment]
@hyp @egorzhdan this is ready for review now.

**Review by egorzhdan — APPROVED** [review]
LGTM.
You'll probably need something like `using ArrayRefOfDiagnosticArgument = ArrayRef<DiagnosticArgument>` to make sure SwiftCompilerSources still compile.

**Comment by zoecarver** [comment]
@swift-ci please test

**Comment by zoecarver** [comment]
@swift-ci please test

File: test/Interop/CxxToSwiftToCxx/bridge-cxx-struct-back-to-cxx-execution.cpp [hunk 1]
```diff
@@ -1,9 +1,9 @@
 // RUN: %empty-directory(%t)
```
Comment (zoecarver) [inline test/Interop/CxxToSwiftToCxx/bridge-cxx-struct-back-to-cxx-execution.cpp:1]: @hyp I'll let you decide if you want to land this PR before fixing this test and the next one. Sorry I did such a terrible job "fixing" these. I couldn't always understand the error, so maybe we can work on them offline.

**Comment by zoecarver** [comment]
@swift-ci please test

**Comment by zoecarver** [comment]
@swift-ci please test

**Review by hyp — APPROVED** [review]

**Pull request merged**